File: Java/1167. Minimum Cost to Connect Sticks.java
class Solution {
    public int connectSticks(int[] sticks) {
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        int cost = 0;
        
        for(int s: sticks){
            heap.add(s);
        }
        
        while(heap.size() > 1){
            int first = heap.poll();
            int second = heap.poll();
            cost += first+second;
            heap.add(first+second);
        }
        
        return cost;
    }
}

File: Python3/1167. Minimum Cost to Connect Sticks.py
class Solution:
    def connectSticks(self, sticks: List[int]) -> int:
        # pop two sticks from the heap
        # add them up and push the sum back to the heap
        # stop condition: len(heap) == 1
        heapq.heapify(sticks)
        cost = 0
        
        while len(sticks) > 1:
            first = heapq.heappop(sticks)
            second = heapq.heappop(sticks)
            #print("first = %d, second = %d" %(first, second))
            cost += first+second
            heapq.heappush(sticks, first+second)
        
        return cost
        